code/ecdsap256/Hacl.P256.fsti

```diff
@@ -258,16 +311,42 @@ val verify_q:
       )
     )
 
+[@@ Comment "Validate a private key.
+
+Input: scalar: uint8[32].
+  \n Output: bool, where true stands for the scalar to be more than 0 and less than order."]
+val validate_private_key: x: lbuffer uint8 (size 32) -> Stack bool
+  (requires fun h -> live h x)
+  (ensures  fun h0 r h1 -> modifies0 h0 h1 /\
+    (
+      let scalar = nat_from_bytes_be (as_seq h0 x) in 
+      r <==> (scalar > 0 && scalar < prime_p256_order)
+    )
+  )
 
-[@ (Comment " There and further we introduce notions of compressed point and not compressed point. 
-  \n We denote || as byte concatenation. 
-  \n A compressed point is a point representaion as follows: (0x2 + y % 2) || x.
-  \n A not Compressed point is a point representation as follows: 0x4 || x || y.
+[@@ CPrologue "
+/*****************************************/
+/* Point representations and conversions */
+/*****************************************/
+
+/*
+  Elliptic curve points have 2 32-byte coordinates (x, y) and can be represented in 3 ways:
+
+  - \"raw\" form (64 bytes): the concatenation of the 2 coordinates, also known as \"internal\"
+  - \"compressed\" form (33 bytes): the first byte is either \x02 or \x03, followed by x
+  - \"uncompressed\" form (65 bytes): the first byte is always \04, followed by the \"raw\" form
+
+  For all of the conversation functions below, the input and output MUST NOT overlap.
+*/
+";
+ (Comment "Convert 65-byte uncompressed to raw.
+
+This function effectively strips the first byte of the uncompressed representation.
```

Comment:
It also checks that the representation is correct.

Reply:
What does it mean for the representation to be correct? That the `x` coordinate does not exceed the order of the group? Does it perform a full call to `validate_public_key`? It would be good to be precise here. I've added a comment but if you could clarify what you mean that'd be great.

Reply:
We call the representation to be correct if the first byte is 0x4/0x2 or 0x3. Probably, it's not so important

Reply:
Ok, but looking at the implementation of decompressionCompressedForm (now known as `compressed_to_raw`), there is also a check `lessThanPrime`... should that be mentioned as one of the possible reasons for failure, too?